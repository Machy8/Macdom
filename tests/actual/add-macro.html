title1 Some text in the h1 element
passwordInput user12345