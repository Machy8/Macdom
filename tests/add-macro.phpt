<?php

use Tester\Assert;


require __DIR__ . '/bootstrap.php';

$inputFunction = function ($line) {
	return '<input type="password" data-user="' . $line . '" placeholder="New password">';
};

$macdom->setup->addMacros = [
	'title1' => function ($line) {
		return '<h1>' . $line . '</h1>';
	},
	'passwordInput' => $inputFunction
];

Assert::matchFile(
	EXPECTED . "/add-macro.html",
	$macdom->compile(file_get_contents(ACTUAL . "/add-macro.html"))
);