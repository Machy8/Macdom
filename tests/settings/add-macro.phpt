<?php

use Tester\Assert;

require '../bootstrap.php';

$testedA = 'title1 Some text in the h1 element';
$resultA = '<h1>Some text in the h1 element</h1>';

$testedB = 'passwordInput user12345';
$resultB = '<input type="password" data-user="user12345" placeholder="New password" />';

$inputFunction = function ($line) {
	return '<input type="password" data-user="' . $line . '" placeholder="New password" />';
};

$o->setup->addMacros = [
	'title1' => function ($line) {
		return '<h1>' . $line . '</h1>';
	},
	'passwordInput' => $inputFunction
];

Assert::same($resultA, $o->compileContent($testedA));
Assert::same($resultB, $o->compileContent($testedB));
