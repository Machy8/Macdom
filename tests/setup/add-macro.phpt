<?php

use Tester\Assert;

require '../bootstrap.php';

$testedA = 'title1 Some text in the h1 element';
$resultA = '<h1>Some text in the h1 element</h1>';

$testedB = 'passwordInput user12345';
$resultB = '<input type="password" data-user="user12345" placeholder="New password">';

$inputFunction = function ($line) {
	return '<input type="password" data-user="' . $line . '" placeholder="New password">';
};

$o->setup->addMacros = [
	'title1' => function ($line) {
		return '<h1>' . $line . '</h1>';
	},
	'passwordInput' => $inputFunction
];

$o->setup->compressCode = TRUE;

Assert::same($resultA, $o->compile($testedA));
Assert::same($resultB, $o->compile($testedB));
