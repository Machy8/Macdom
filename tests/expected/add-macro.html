<h1>Some text in the h1 element</h1>
<input type="password" data-user="user12345" placeholder="New password">